into/backends/tests/test_aws.py

```diff
@@ -40,8 +44,38 @@
 js = pd.io.json.loads(pd.io.json.dumps(df, orient='records'))
 
 
+is_authorized = False
+tried = False
+
+
 @pytest.fixture
 def db():
+    # if we aren't authorized and we've tried to authorize then skip, prevents
+    # us from having to deal with timeouts
+
+    # TODO: this will fail if we want to use a testing cluster with a different
+    # security group than 'default'
+    global is_authorized, tried
+
+    if not is_authorized:
+        if not tried:
+            public_ip = json.load(urlopen('http://httpbin.org/ip'))['origin']
+            conn = boto.connect_redshift()
+            ip = public_ip + '/32'
+            try:
+                conn.authorize_cluster_security_group_ingress('default',
+                                                              cidrip=ip)
+            except boto.redshift.exceptions.AuthorizationAlreadyExists:
+                is_authorized = True
+            except Exception as e:
+                pytest.skip('authorization to access redshift cluster failed %s' %
+                            e)
+            else:
+                is_authorized = True
+            finally:
+                tried = True
+        else:
+            pytest.skip('authorization to access redshift cluster failed')
```

Comment:
This is kind of gross, but for testing i wanted to automate adding whatever my current ip address was to the default cluster security group. For example, if I get assigned a new ip address I don't want to have to log in to the AWS console and add my current address to the security group. This also allows us to determine whether or not someone will be able to access the cluster at all, and therefore we don't have to depend on timing out to determine whether or not we can connect.